model/common/src/icon4py/model/common/grid/base.py

```diff
@@ -138,7 +138,7 @@ def _get_offset_provider(self, dim, from_dim, to_dim):
             dim, self.connectivities[dim].dtype
         )
         return gtx.NeighborTableOffsetProvider(
-            xp.asarray(self.connectivities[dim]),
+            gtx.as_field([from_dim, gtx.Dimension("neigh", kind=gtx.DimensionKind.LOCAL)], xp.asarray(self.connectivities[dim])).ndarray,
```

Comment:
This is just a hack to get the memory layout that the backend likes (horizontal with stride 1 for gpu). The proper fix should probably use `gtx.as_connectivity()` somewhere. GT4Py currently doesn't really help to get this clean...